### PR TITLE
Standardise on the term record encryption

### PR DESCRIPTION
**Pull request by SamBarker**
### Type of change

- Refactoring

### Description

We believe the term `Record Encryption` is clearer, easier to understand for users  and a more accurate description of what it does than `Topic Encryption`. So to avoid ambiguity this PR standardises on the term `recordEncryption`.

### Additional Context

While `EnvelopeEncryption` is an accurate description of our implementation it feels odd to use one term to describe the feature and then a completely different one for the filter itself. From the work we have done it also doesn't feel realistic to believe we will support alternative implementations of `Record Encryption`. Additionally we may also use envelope encryption to support field level encryption and crypto shredding use cases but those would be independent filters as well.

### Checklist

_Please go through this checklist and make sure all applicable tasks have been done_

- [ ] Write tests
- [ ] Make sure all tests pass
- [ ] Review performance test results. Ensure that any degradations to performance numbers are understood and justified.
- [ ] Make sure all Sonarcloud warnings are addressed or are justifiably ignored.
- [ ] Update documentation
- [ ] Reference relevant issue(s) and close them after merging
- [ ] For user facing changes, update CHANGELOG.md (remember to include changes affecting the API of the test artefacts too).

### CODE REVIEW

**Comment by SamBarker** [comment]
This branch just moves some code around none of the issues sonar is highlighting are new. How do people feel about merging without a green sonar build?

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,7 @@ Please enumerate **all user-facing** changes using format `<githib issue/pr numb
 
 ## 0.5.0
 
+* [#1043](https://github.com/kroxylicious/kroxylicious/pull/1043): Rename EnvelopeEncryption filter to RecordEncryption
```
Comment (k-wall) [inline CHANGELOG.md:7]: Do we need to call this out in the changes section?

**Review by k-wall — APPROVED** [review]
lgtm

**Comment by k-wall** [comment]
> This branch just moves some code around none of the issues sonar is highlighting are new. How do people feel about merging without a green sonar build?

Let's merge.

File: CHANGELOG.md (hunk 1 above)
Comment (SamBarker) [reply CHANGELOG.md:7]: I've added that.

**Pull request merged**